sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -422,7 +422,7 @@ object FunctionRegistry {
     expression[BitwiseXor]("^"),
 
     // json
-    expression[StructToJson]("to_json"),
+    expression[StructOrArrayToJson]("to_json"),
```

Comment:
It seems `StructOrArrayToJson` is a little ambiguous cuz this `Array` means an array of structs, right? `StructsToJson` is better?